etcdserver/v3demo_server.go

```diff
@@ -300,12 +302,12 @@ func applyRange(txnID int64, kv dstorage.KV, r *pb.RangeRequest) (*pb.RangeRespo
 	}
 
 	if txnID != noTxn {
-		kvs, rev, err = kv.TxnRange(txnID, r.Key, r.RangeEnd, limit, r.Revision)
+		kvs, _, err = kv.TxnRange(txnID, r.Key, r.RangeEnd, limit, r.Revision)
```

Comment:
After I took a more careful look I found the root cause is that TxnRange or Range should return current rev not the rev . So we should fix storage. And keep this file as it is.